binding/dotnet/PvSpeaker/PvSpeaker.cs

```diff
@@ -0,0 +1,441 @@
+            }
+
+            byte[] utf8Bytes = Encoding.UTF8.GetBytes(outputPath + '\0');
+            IntPtr outputPathPtr = Marshal.AllocHGlobal(utf8Bytes.Length);
+            Marshal.Copy(utf8Bytes, 0, outputPathPtr, utf8Bytes.Length);
+
+            pv_speaker_write_to_file(_libraryPointer, outputPathPtr);
+        }
+
+        /// <summary>
+        /// Gets whether the speaker has started and is available to receive PCM data or not.
+        /// </summary>
+        public bool IsStarted
+        {
+            get
+            {
+                return pv_speaker_get_is_started(_libraryPointer) != 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets the sample rate matching the value passed to the constructor.
+        /// </summary>
+        public int SampleRate
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Gets the bits per sample matching the value passed to the constructor.
+        /// </summary>
+        public int BitsPerSample
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Gets the buffer size in seconds matching the value passed to the constructor.
+        /// </summary>
+        public int BufferSizeSecs
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Gets the current selected audio device.
+        /// </summary>
+        public string SelectedDevice
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Gets the current version of the library.
+        /// </summary>
+        public string Version
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Gets a list of the available audio output devices on the current system.
+        /// </summary>
+        /// <returns>An array of strings containing the names of the audio devices.</returns>
+        public static string[] GetAvailableDevices()
+        {
+            PvSpeakerStatus status = pv_speaker_get_available_devices(out int deviceListLength, out IntPtr deviceList);
+            if (status != PvSpeakerStatus.SUCCESS)
+            {
+                throw PvSpeakerStatusToException(status);
+            }
+
+            int elementSize = Marshal.SizeOf(typeof(IntPtr));
+            string[] deviceNames = new string[deviceListLength];
+            for (int i = 0; i < deviceListLength; i++)
+            {
+                deviceNames[i] = Marshal.PtrToStringAnsi(Marshal.ReadIntPtr(deviceList, i * elementSize));
+            }
+
+            pv_speaker_free_available_devices(deviceListLength, deviceList);
+
+            return deviceNames;
+        }
+
+        /// <summary>
+        /// Converts status codes to PvSpeakerExceptions.
+        /// </summary>
+        /// <param name="status">Status code.</param>
+        /// <returns>PvSpeakerExceptions</returns>
+        private static PvSpeakerException PvSpeakerStatusToException(PvSpeakerStatus status)
```

Comment:
This should probably take an optional message that can be thrown with the exception